src/object/cli_obj.c

```diff
@@ -2414,6 +2478,10 @@ do_dc_obj_fetch(tse_task_t *task, daos_obj_fetch_t *args,
 	if (rc != 0)
 		D_GOTO(out_task, rc);
 
+	rc = csum_obj_fetch(obj, args);
```

Comment:
same as update case that only need to calculate the key csum when non-retry case right? (if (!obj_auxi->io_retry))